### ISSUE TITLE
Generator + default argument w/ variable reference generates bad code

### ISSUE THREAD

**Issue by monsanto** [6to5/6to5#541]
```
let foo = 0
let x = function*(test=foo) {
    yield test
}

console.log(x().next().value)
```

output: `{ next: [Function], throw: [Function], return: [Function] }`

change `test=foo` to `test=0` and you get `0` as output as expected.

**Comment by sebmck** [comment]
There have been so really weird generator issues such as #438 should be easier for 3.0.0 with [regenerator-6to5](https://github.com/6to5/regenerator-6to5).

**Comment by sebmck** [comment]
Fixed via da8888a3507ebd19e1d5dd3391340747b21d30f8 in the 3.0.0 branch.

**Comment by sebmck** [comment]
Fixed as of 3.0.0
